common/src/main/java/io/druid/math/expr/Function.java

```diff
@@ -792,4 +874,172 @@ public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
       return eval.isNull() ? args.get(1).eval(bindings) : eval;
     }
   }
+
+  class ConcatFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "concat";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() == 0) {
+        return ExprEval.of(null);
+      } else {
+        final StringBuilder builder = new StringBuilder(Strings.nullToEmpty(args.get(0).eval(bindings).asString()));
+        for (int i = 1; i < args.size(); i++) {
+          final String s = args.get(i).eval(bindings).asString();
+          if (s != null) {
+            builder.append(s);
+          }
+        }
+        return ExprEval.of(builder.toString());
+      }
+    }
+  }
+
+  class StrlenFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "strlen";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 1) {
+        throw new IAE("function '%s' needs 1 argument", name());
+      }
+
+      final String arg = args.get(0).eval(bindings).asString();
+      return arg == null ? ExprEval.of(0) : ExprEval.of(arg.length());
+    }
+  }
+
+  class SubstringFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "substring";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 3) {
+        throw new IAE("function '%s' needs 3 arguments", name());
+      }
+
+      final String arg = args.get(0).eval(bindings).asString();
+
+      // Behaves like SubstringDimExtractionFn, not SQL SUBSTRING
+      final int index = args.get(1).eval(bindings).asInt();
+      final ExprEval lengthExprEval = args.get(2).eval(bindings);
+      final int end = lengthExprEval.asInt() >= 0 ? index + lengthExprEval.asInt() : -1;
+
+      if (arg == null) {
+        return ExprEval.of(null);
+      }
+
+      if (index < arg.length()) {
+        if (end > 0) {
+          return ExprEval.of(arg.substring(index, Math.min(end, arg.length())));
+        } else {
+          return ExprEval.of(arg.substring(index));
+        }
+      } else {
+        return ExprEval.of(null);
+      }
+    }
+  }
+
+  class ReplaceFunc implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "replace";
+    }
+
+    @Override
+    public ExprEval apply(List<Expr> args, Expr.ObjectBinding bindings)
+    {
+      if (args.size() != 3) {
+        throw new IAE("function '%s' needs 3 arguments", name());
+      }
+
+      final String arg = args.get(0).eval(bindings).asString();
+      final String pattern = args.get(1).eval(bindings).asString();
+      final String replacement = args.get(2).eval(bindings).asString();
+      return ExprEval.of(
```

Comment:
`ExprEval.of` will do this. It includes `Strings.emptyToNull(value)` in its constructor.